gamestonk_terminal/cryptocurrency/onchain/whale_alert_model.py

```diff
@@ -0,0 +1,124 @@
+"""Whale Alert model"""
+__docformat__ = "numpy"
+
+from typing import Optional
+import textwrap
+import requests
+import pandas as pd
+import numpy as np
+import gamestonk_terminal.config_terminal as cfg
+
+
+class ApiKeyException(Exception):
+    """Api Key Exception object"""
+
+    def __init__(self, message: str):
+        super().__init__(message)
+        self.message = message
+
+    def __str__(self) -> str:
+        return "ApiKeyException: %s" % self.message
+
+
+def make_request(params: Optional[dict] = None) -> dict:
+    """Helper methods for requests [Source: https://docs.whale-alert.io/]
+
+    Parameters
+    ----------
+    params: dict
+        additional param
+
+    Returns
+    -------
+    dict:
+        response from api request
+    """
+
+    api_key = cfg.API_WHALE_ALERT_KEY or ""
+    url = "https://api.whale-alert.io/v1/transactions?api_key=" + api_key
+    response = requests.get(url, params=params)
+
+    if not 200 <= response.status_code < 300:
+        raise ApiKeyException("Invalid Authentication: %s" % response.text)
+    try:
+        return response.json()
+    except Exception as e:
+        raise ValueError("Invalid Response: %s" % response.text) from e
+
+
+def get_whales_transactions(min_value: int = 800000, limit: int = 100) -> pd.DataFrame:
+    """Whale Alert's API allows you to retrieve live and historical transaction data from major blockchains.
+    Supported blockchain: Bitcoin, Ethereum, Ripple, NEO, EOS, Stellar and Tron. [Source: https://docs.whale-alert.io/]
+
+    Parameters
+    ----------
+    min_value: int
+        Minimum value of trade to track.
+    limit: int
+        Limit of transactions. Max 100
+
+    Returns
+    -------
+    pd.DataFrame
+        Crypto wales transactions
+    """
+
+    min_value = 800000 if min_value > 800000 else min_value
```

Comment:
A more elegant way to do this is using max, i.e.
```
min_value = max(min_value, 800000)
limit = max(limit, 100)
```

And if you want to limit a number between a and b, where a < b, you can do:
```
number =max(min(b, number), a)
```

Reply:
Cool :) I knew this way for that, but just I had this kind of "bad habbit" of always doing things with `x if x else y` . But will try to have this elegant way in mind and implement it in the code!